receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -74,24 +79,36 @@ func newReceiver(params receiver.Settings, config *Config, consumer consumer.Log
 func (kr *k8sobjectsreceiver) Start(ctx context.Context, _ component.Host) error {
 	client, err := kr.config.getDynamicClient()
 	if err != nil {
-		return err
+		return kr.handleError(err, "failed to create dynamic client")
 	}
 	kr.client = client
 
 	// Validate objects against K8s API
 	validObjects, err := kr.config.getValidObjects()
 	if err != nil {
-		return err
+		return kr.handleError(err, "failed to get valid objects")
 	}
 
+	var validConfigs []*K8sObjectsConfig
 	for _, object := range kr.objects {
 		gvrs, ok := validObjects[object.Name]
 		if !ok {
-			availableResource := make([]string, len(validObjects))
-			for k := range validObjects {
-				availableResource = append(availableResource, k)
+			err := fmt.Errorf("resource not found: %s", object.Name)
+			// First handle the error which will log it if in ignore mode
+			if handlerErr := kr.handleError(err, ""); handlerErr != nil {
+				return handlerErr
+			}
+
+			// Then log additional context if in ignore mode
+			if kr.config.ErrorMode == IgnoreError {
+				availableResource := make([]string, 0, len(validObjects))
+				for k := range validObjects {
+					availableResource = append(availableResource, k)
+				}
+				kr.setting.Logger.Warn("Available resources in cluster",
```

Comment:
Can we just pass the `Available resources in cluster...` message to `handleError` above? Right now we log twice for the same reason, right?